stacker/providers/aws/default.py

```diff
@@ -94,6 +94,9 @@ def __init__(self, region, **kwargs):
         # Necessary to deal w/ multiprocessing issues w/ sharing ssl conns
         # see: https://github.com/remind101/stacker/issues/196
         self._pid = os.getpid()
+        # If region is not set explicitly, fetch it from the boto session
+        if not self.region:
```

Comment:
I think typical usage of get_session() was without any region being passed in; the only exception is if a user passes the `--region` argument. Otherwise, the region is set by the credential chain; that is, it uses boto default session logic to resolve the region, which amounts to AWS_REGION env variable, or the contents of the profile (`~/.aws/credentials | config`).

I did see the comment about "not clobbering existing region" but that's not a good idea. The only way someone would already have region in the environment is if they passed it in. For example, someone could: `stacker build -r us-east-1 -e region=us-west-2`. This will establish a boto session to `us-east-1` but use `us-west-2` in the template - Stacker shouldn't allow that to happen.

I'll look at adding a test case specific to this change.